format.sh

```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+git ls-files | grep -E ".*\.ml(i)?$" | xargs -I% bash -c 'ocp-indent -i %'
```

Comment:
This would reformat _all_ the files in the repository. It seems `ocp-indent` will rewrite every file even if reformatting does nothing. One could use `git ls-files --modified` to reformat only the modified files. Also, why do we need `bash`? (Maybe it's necessary for macOS?)

The following only reformats the files that will be committed. This would work if we run it every time.
```suggestion
git diff --name-only | grep -E ".*\.ml(i)?$" | xargs -I% ocp-indent -i %
```

Reply:
Can you explain why we shouldn't reformat all the files? I prefer that behavior to reformatting only the ones to be committed, because it ensures that we keep things well-formatted even if someone forgot to run it.

But I'm open to an argument that this is bad behavior if you think it is.

Reply:
I think we could have both. The points against reformatting all files are:
1. It clutters the git history---for example this PR now has unrelated changes. We can use `git diff origin/main --name-only` so that it will calculate all the changes files with respect to the main branch. We can even set up a CI that rejects a PR which introduces new files which are not formatted. This should prevent the trouble you mentioned.
2. Currently, `ocp-indent` will rewrite every single file it processes even if nothing has changed. 95% of the files are already perfectly formatted and so the original script is doing like 20x useless work.

Reply:
(1) is reasonable.  But (2) is not important, because it complete instantly regardless.

Reply:
I slightly prefer an efficient yet still simple solution that can be recycled for future projects, but yes, for tiny projects like cooltt on a machine with SSD it makes no differences. Anyways, it seems you are fine with
```suggestion
git diff origin/main --name-only | grep -E ".*\.ml(i)?$" | xargs -I% ocp-indent -i %
```
along with a CI to reject unruly PRs?

Reply:
Personally I don't care if the CI rejects unruly PRs --- I think this would be annoying because something might not be formatted perfectly in the middle of development, but I do want to see the CI result. But obviously I am fine with including your script (let's have both).